### ISSUE TITLE
Draw icon on top of gauge but below the needle?

### ISSUE THREAD

**Issue by Moulde** [Mikhus/canvas-gauges#82]
Hi
I'm trying to recreate a gauge like this.
http://www.publicdomainpictures.net/pictures/50000/velka/fuel-gauge.jpg

So i need to draw an icon on the gauge.
I tried just getting the canvas context and drawing the icon, and it works, but after updating the value, the gauge is redrawn again, and so the image disappears.

And i am not able to draw the icon below the needle, as it needs to be drawn in the correct order.

Any ideas?
Thanks in advance.

**Comment by Mikhus** [comment]
Hi. Thank you for contribution!

Yes, I have an idea on this - is to provide some events which will make possible to inject into gauge drawing order. Something like:

```javascript
gauge.beforeDraw('needle', function(canvasContext) {
   // do some custom drawings on a given canvas
});
```

But this currently is not implemented, will have it as a working idea for further releases.

**Comment by Moulde** [comment]
That sounds like a great way to implement it.

**Comment by Mikhus** [comment]
Let's keep it open until implemented :)

**Comment by Mikhus** [comment]
Implemented in branch 2.1.0

**Comment by ashokkumarmr1** [comment]
Where  and how do i call guage.before()? 
I added guage.js inside asset folder
